source/utilities.cc

```diff
@@ -944,34 +961,157 @@ namespace aspect
+                                               "delivers correct data."));
+                    }
+                }
+
+              // Add the POINTS data that is required and found at the top of the data file.
+              // The POINTS values are set as attributes inside a table.
+              // Loop through the Attribute table to locate the points values within
+              std::vector<std::string> points;
+              for (libdap::AttrTable::Attr_iter i = das.var_begin(); i != das.var_end(); i++)
+                {
+                  libdap::AttrTable *table;
+
+                  table = das.get_table(i);
+                  if (table->get_attr("points") != "")
+                    points.push_back(table->get_attr("points"));
+                }
+
+              std::stringstream urlString;
+
+              // Append the gathered POINTS in the proper format:
+              // "# POINTS: <val1> <val2> <val3>"
+              urlString << "# POINTS:";
+              for (unsigned int i = 0; i < points.size(); i++)
+                {
+                  urlString << " " << points[i];
+                }
+              urlString << "\n";
+
+              //Add the values from the arrays into the stringstream. The values are passed in
+              // per row with a character return added at the end of each row.
+              // TODO: Add a check to make sure that each column is the same size before writing
+              //     to the stringstream
+              for (unsigned int i = 0; i < tmp.size(); i++)
+                {
+                  for (unsigned int j = 0; j < columns.size(); j++)
+                    {
+                      urlString << columns[j][i];
+                      urlString << " ";
+                    }
+                  urlString << "\n";
+                }
+
+              data_string = urlString.str();
+              filesize = data_string.size();
+
+              delete url;
+#else // HAVE_LIBDAP
 
-          if (!filestream.eof())
-            {
               // broadcast failure state, then throw
-              MPI_Bcast(&filesize,1,MPI_UNSIGNED,0,comm);
-              AssertThrow (false,
-                           ExcMessage (std::string("Reading of file ") + filename + " finished " +
-                                       "before the end of file was reached. Is the file corrupted or"
-                                       "too large for the input buffer?"));
-              return data_string; // never reached
+              MPI_Bcast(&filesize, 1, MPI_UNSIGNED, 0, comm);
+              AssertThrow(false,
+                          ExcMessage(std::string("Reading of file ") + filename + " failed. " +
+                                     "Make sure you have the dependencies for reading a url " +
+                                     "(when running cmake make sure -DLIBDAP_ON=ON)"));
+
+              std::ifstream filestream(filename.c_str());
```

Comment:
why are you opening a file here? Isn't this the case where it is a URL and we don't have libdap? I think you can delete this whole block of code.

Reply:
I think you're right, fixed that.